### DISCUSSION TITLE
I'm not a dev of any kind,, but want more axis on vjoy,,

### DISCUSSION THREAD

**Discussion by tsmspace** [njz3/vJoy#6] [General]
Is it possible to raise the supported number of axis to 12? 
I need 4 joysticks to control spacecraft in games, and want to raise my stick count so that the same controller has throttle for aerodynamic games.

this project is based on vjoyserialfeeder and the original 4 joystick project is viewable at tsmspace.com

![20221022_171114](https://user-images.githubusercontent.com/45222389/197373665-be1816ec-e265-4e8e-b579-5c02bcc1a0d6.jpg)
![20221022_171119](https://user-images.githubusercontent.com/45222389/197373667-543c823a-2414-4938-ae00-fdfa578d64f3.jpg)
![20221022_115813](https://user-images.githubusercontent.com/45222389/197373668-c06d0519-afea-4fbd-8a63-4effa64b293e.jpg)

I have a video showing the project here,, 

 https://youtu.be/h0K4BEeDUJY

**Comment by unhappytroll** [comment]
iirc, Windows only allows 8 axis per joystick. just add second VJoy device.

**Reply by tsmspace** [reply]
edit::::: I haven't found where I can split my vjoyserialfeeder output to multiple vjoy inputs. I have found where multiple device outputs can be combined into a single vjoy input, but not the other way around. vjoyserialfeeder allows me to assign which axis goes where, and I can remap sticks to work as buttons at the stick extremes, so it's possible to do some amount of remapping that way to make use of the sticks,, but presently I don't see how to get actual stick axis for all 12 axis. (ultimately I want at least 14, but this prototype is 12). I am hoping to contact microsoft support and with my one little question to change the direction of the company to accept up to 16 axis...... maybe it will go well. 

edit 2 :::::: it will be possible to map (in vjoyserialfeeder) the new joysticks to register as 4 buttons, one for each direction, so it will be possible to gain some degree of utility from the sticks, although not the full utility that I desire. ,,, all is not lost, and my project is justified to continue, but hopefully in the long term there will be the desired solution of windows supporting more axis. (and then of course vjoy supporting them afterwards as well)
you mean I can have the vjoyserialfeeder output go to two separate vjoy devices? I'm not sure how to do that,, and then not sure if my games will support having multiple joysticks at once. 

I will look for tutorials.

**Reply by unhappytroll** [reply]
iirc, VKB configuration software now can create multiple virtual joysticks inside one device. may be you should do something like that, because I really doubt MS will change that - 8 axis/32 buttons is very old legacy code...

**Reply by tsmspace** [reply]
I could try to somehow campaign for people to upvote a feedback requesting it to change
-- vkb looks to be specific to their own hardware. 
I saw that joystick gremlin might be able to map it's own configuration to multiple virtual devices, but vjoyserialfeeder only feeds one virtual device, and then vjoy won't pick up all the axis, and that would need to happen first before going to joystick gremlin,, and of course finally I don't know that my games will support several joysticks simultaneously.

**Comment by cyberluke** [comment]
What game is this for?

**Reply by tsmspace** [reply]
It's for space simulator games in general but then also to support quadcopter controls on the same device. The layout will be like an old nintendo64 controller where there are two spots for your left hand. Then, the use case where a single controller is desired would be that you are flying your ship in a realistic re-entry, land, and then launch a quadcopter drone to perform a mission of some kind. The other use case is a game where you fly a space-ship for a realistic re-entry, and then instead of flying on the rocket, the ship is equipped with a jet that then makes it better if you can set the throttle with a floating throttle stick (like an rc quadcopter throttle). So, the idea is a gamepad that works well in both space and flight simulators, particularly games that do both.

**Reply by cyberluke** [reply]
No in general. You must be specific. What game is this for?

I have been solving something similar on my Github (I have forked this vJoy and FreePIE).

**Reply by cyberluke** [reply]
I'm game developer and I really don't know what you are talking about :-D you need to show some game and then other developers can help here.

For example I use FreePie to create one vJoy device out of 3 hardware joysticks. Same way you could use one hardware device to output to 3 separate vJoy devices and route it inside FreePie based on a communication protocol of your choice.

**Reply by cyberluke** [reply]
For example Descent supports two joysticks out of the box. That is 6-DOF movement for space sims. That is nothing new.

**Reply by cyberluke** [reply]
Your idea is a custom controller with custom game. That is not 'space simulator games in general'. Normal commercial developer would use one stick for both quadcopter and ship and use some flip switch (or button). Or there would be some semi-autopilot with balance mode for spaceship. 

Even your idea seems to be silly and I have not seen any game that would use it, and you probably too, because you did not mention it, you can still do it using 2 or 3 vJoy devices.

For example I have USB Logitech G940 (flight stick, rudder, pedals). It is connected via one USB cable to computer, but then it will show up as 3 separate gaming devices. The disadvantage is that only two games on the market fully supports it out of the box.

**Reply by tsmspace** [reply]
> Even your idea seems to be silly and I have not seen any game that would use it, and you probably too, because you did not mention it, you can still do it using 2 or 3 vJoy devices.
> 

I'm really hoping to have it all on one single device, even while most of the applications would be separate games. Also supporting more than 8 axis is already a need in normal 6dof games because I want to use camera look on it's own independent and simultaneously live stick. Some (most playtime) the games that I play are Infinity Battlescape, Flight of Nova, Fly Dangerous, Astrokill, Kerbal Space Program, Orbital Racer, Starmade, and my own homemade game (actually originally made by Why485) MicroGravityDreams. I used to play a lot of FPV simulators, and have been considering normal flight sims like Microsoft Flight simulator. So again one of the primary intents is a handheld gamepad with two spots for the left hand, space and air, and again the intent is for this to be a single device with all controls simultaneously live. Finally the intent is for the project to be very simple to reproduce from some online instructions (that others may duplicate the project with just some files on the project page).
